sklearn/utils/validation.py

```diff
@@ -876,7 +1012,7 @@ def check_array(
                         )
                     array = xp.astype(array, dtype, copy=False)
                 else:
-                    array = _asarray_with_order(array, order=order, dtype=dtype, xp=xp)
+                    array = asarray(array, order=order, dtype=dtype)
```

Comment:
For the third party array libraries you are considering, do they usually work with NumPy dtypes, such as `np.float32`?